src/server.rs

```diff
@@ -320,7 +320,11 @@ impl Service for PushHandler {
                         warn!("Error: {}", e);
                         let body = "Push not successful";
                         future::ok(Response::builder()
-                            .status(StatusCode::INTERNAL_SERVER_ERROR)
+                            .status(if let SendPushError::ProcessingClientError(_) = e {
+                                StatusCode::BAD_REQUEST
+                            } else {
+                                StatusCode::INTERNAL_SERVER_ERROR
```

Comment:
Perhaps `502`? I think `500` should be reserved for actual unexpected errors (such as exceptions) in the relay server.

Reply:
The term *unexpected* is very loose. :stuck_out_tongue_closed_eyes: